monai/transforms/spatial/array.py

```diff
@@ -560,6 +575,16 @@ def __call__(
         out_d = self.pixdim[:sr]
         if out_d.size < sr:
             out_d = np.append(out_d, [1.0] * (sr - out_d.size))
+        orig_d = affine_to_spacing(affine_, sr, out_d.dtype)
+        for idx, (_d, mn, mx) in enumerate(
+            zip_longest(orig_d, self.min_pixdim[:sr], self.max_pixdim[:sr], fillvalue=np.nan)
+        ):
+            target = out_d[idx]
+            mn = target if np.isnan(mn) else min(mn, target)
+            mx = target if np.isnan(mx) else max(mx, target)
+            if mn > mx:
+                raise ValueError(f"min_pixdim is larger than max_pixdim at dim {idx}: min {mn} max {mx} out {target}.")
+            out_d[idx] = _d if (mn - AFFINE_TOL) <= _d <= (mx + AFFINE_TOL) else target
```

Comment:
1) if we define min and max pixdim, why do we need additional affine_tol here?
2) if the resolution is withing the min / max, we simply want to skip the resampling completely. In this code, it seems all it does is records the original resolution and still does resampling in the next step (which will incur additional compute load and possible artifacts)